diff -Nuarp origin/_deps/flatbuffers-src/include/flatbuffers/base.h fixed/_deps/flatbuffers-src/include/flatbuffers/base.h
--- _deps/flatbuffers-src/include/flatbuffers/base.h	2024-07-15 18:46:49.285744238 +0800
+++ _deps/flatbuffers-src/include/flatbuffers/base.h	2024-07-16 09:13:24.000000000 +0800
@@ -270,7 +270,7 @@ namespace flatbuffers {
   // strtoull_l}.
   #if (defined(_MSC_VER) && _MSC_VER >= 1800) || \
       (defined(__ANDROID_API__) && __ANDROID_API__>= 21) || \
-      (defined(_XOPEN_VERSION) && (_XOPEN_VERSION >= 700)) && \
+      (defined(GLIBC) && defined(_XOPEN_VERSION) && (_XOPEN_VERSION >= 700)) && \
         (!defined(__Fuchsia__) && !defined(__ANDROID_API__))
     #define FLATBUFFERS_LOCALE_INDEPENDENT 1
   #else
